### PR TITLE
Add pyxtal wrapper

### PR DESCRIPTION
**Pull request by pmrv**

### CODE REVIEW

**Comment by jan-janssen** [comment]
`center_coordinates_in_unit_cell` is not an ASE compatible command: 
```
    s.center_coordinates_in_unit_cell()
AttributeError: 'Atoms' object has no attribute 'center_coordinates_in_unit_cell'
```
But you can use the `structuretoolkit.common.center_coordinates_in_unit_cell()` function to achieve the same.

File: structuretoolkit/build/pyxtal.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from typing import Union, List, Tuple
+import warnings
+
+from ase import Atoms
+from tqdm.auto import tqdm
+from pyxtal import pyxtal as _pyxtal
+from pyxtal.msg import Comp_CompatibilityError
+
```
Comment (jan-janssen) [inline structuretoolkit/build/pyxtal.py:12]:
```suggestion
from structuretoolkit.common.helper import center_coordinates_in_unit_cell


```

File: structuretoolkit/build/pyxtal.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from typing import Union, List, Tuple
+import warnings
+
+from ase import Atoms
+from tqdm.auto import tqdm
+from pyxtal import pyxtal as _pyxtal
+from pyxtal.msg import Comp_CompatibilityError
```
Comment (jan-janssen) [inline structuretoolkit/build/pyxtal.py:11]:
```suggestion
```

File: structuretoolkit/build/pyxtal.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    Returns:
+        :class:`~.Atoms`: the generated structure, if repeat==1 and only one symmetry group is requested
+        list of dict of all generated structures, if repeat>1 or multiple symmetry groups are requested
+
+    Raises:
+        ValueError: if `species` and `num_ions` are not of the same length
+        ValueError: if stoichiometry and symmetry group are incompatible and allow_exceptions==False or only one structure is requested
+    """
+    if len(species) != len(num_ions):
```
Comment (jan-janssen) [inline structuretoolkit/build/pyxtal.py:52]:
```suggestion
    from pyxtal import pyxtal as _pyxtal
    from pyxtal.msg import Comp_CompatibilityError

    if len(species) != len(num_ions):
```

**Comment by jan-janssen** [comment]
One goal of the `structuretoolkit` package is to remain very stream-lined. The default installation via pypi only requires `ase`, `matplotlib`, `numpy` and `spicy`, with the last three being requirements of `ase`. This is enabled by moving the imports into the functions, even though this conflicts with the PEP8 recommendations. Furthermore the tests are skipped when the corresponding packages are not available, this enables the validation of the minimal installation.

**Comment by pmrv** [comment]
[I merged (and rebased) some of your changes before the force push]

> One goal of the `structuretoolkit` package is to remain very stream-lined. The default installation via pypi only requires `ase`, `matplotlib`, `numpy` and `spicy`, with the last three being requirements of `ase`. This is enabled by moving the imports into the functions, even though this conflicts with the PEP8 recommendations. Furthermore the tests are skipped when the corresponding packages are not available, this enables the validation of the minimal installation.

I see the goal, but what about this way?  This only conditionally defines the function and raises a warning if pyxtal is missing, so that users see it early and cannot even attempt to call the missing function.  No strong feeling either, also happy to implement your original solution if you want it consistent everywhere.

Does sound like we want `ImportAlarm` here, though.

File: tests/test_pyxtal.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from unittest import TestCase
```
Comment (pmrv) [inline tests/test_pyxtal.py:1]:
```suggestion
from unittest import TestCase, skipIf
```

File: tests/test_pyxtal.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+
+try:
+    import pyxtal
+    skip_pyxtal_test = False
+except ImportError:
+    skip_pyxtal_test = True
+
+
+@unittest.skipIf(skip_pyxtal_test, "pyxtal is not installed, so the pyxtal tests are skipped.")
```
Comment (pmrv) [inline tests/test_pyxtal.py:13]:
```suggestion
@skipIf(skip_pyxtal_test, "pyxtal is not installed, so the pyxtal tests are skipped.")
```

**Pull request merged**